src/poparser.js

```diff
@@ -169,14 +180,14 @@ Parser.prototype._lexer = function (chunk) {
             value: '',
             quote: chr
           };
-          this._lex.push(this._node);
+          this._lex.push(/** @type {Node} */ (this._node));
```

Comment:
They start off as a `Partial<Node>`, but when they are pushed into the `_lex` they need to be forced into the correct type. This in JSDocs `/** @type {Node} */ (this._node)` is the same as `this._node as Node`.

Reply:
isn't enough to declare `this._lex` as `node[]`?